services/QuillLMS/client/app/bundles/Diagnostic/components/questions/responseList.tsx

```diff
@@ -0,0 +1,130 @@
+import { focusPointMatchHelper, incorrectSequenceMatchHelper } from "quill-marking-logic"
+import * as React from 'react'
+import _ from 'underscore'
+
+import Response from './response'
+
+import { AffectedResponse, isValidRegex } from '../../../Shared/index'
+import massEdit from '../../actions/massEdit'
+import { Question, FocusPoint, IncorrectSequence } from '../../libs/grading/rematching'
+
+interface ResponseListProps {
+  admin: boolean,
+  ascending: boolean,
+  concepts: Array<Object>,
+  dispatch: Function,
+  expand: Function,
+  expanded: boolean,
+  getChildResponses: Function,
+  getResponse: Function,
+  massEditResponses: Object,
+  mode: string,
+  question: Question,
+  questionID: string,
+  responses: Array<Response>,
+  selectedFocusPoints: Array<FocusPoint>,
+  selectedIncorrectSequences: Array<IncorrectSequence>,
+  states: Object,
+  updateResponse: Function,
+}
+
+
+const ResponseList = ({admin, ascending, concepts, dispatch, expand, expanded, getChildResponses, getResponse, massEditResponses, mode, question, questionID, responses, selectedFocusPoints, selectedIncorrectSequences, states, updateResponse }: ResponseListProps) => {
+
+  function allResponsesChecked() {
+    return !responses.some((r) => {
+      return !(
```

Comment:
This double negative is a little confusing to me. I think we could just do `.every` instead of `.some` here without any bang operators, right?